test/applications/cmd/monitoring-dns-nsc/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/connectioncontext"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	"strings"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/monitor"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/monitor/local"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/networkservicemesh/networkservicemesh/sdk/client"
+)
+
+var version string
+
+//TODO: cleanup
+func main() {
+	corefile := NewCorefile("/etc/coredns/Corefile")
+	corefile.WriteScope(".:53").Write("log")
+	logrus.Info("Starting monitoring-dns-nsc...")
+	logrus.Infof("Version: %v", version)
+	// Capture signals to cleanup before exiting
+	c := tools.NewOSSignalChannel()
+
+	tracer, closer := tools.InitJaeger("nsc")
+	opentracing.SetGlobalTracer(tracer)
+	defer func() { _ = closer.Close() }()
+
+	nsc, err := client.NewNSMClient(context.Background(), nil)
+	if err != nil {
+		logrus.Fatalf("Unable to create the NSM client: %v", err)
+	}
+
+	currentConn, err := nsc.Connect("nsm", "kernel", "Primary interface")
+	if err != nil {
+		logrus.Fatalf("Failed to connect: %v", err)
+	}
+
+	logrus.Info("nsm client: initialization is completed successfully.")
+
+	monitorClient, err := local.NewMonitorClient(nsc.NsmConnection.GrpcClient)
+	if err != nil {
+		logrus.Fatalf("Failed to start monitor client: %v", err)
+	}
+	defer monitorClient.Close()
+
+	for {
+		select {
+		case <-c:
+			return
+		case err = <-monitorClient.ErrorChannel():
+			logrus.Fatalf("Monitor failed: %v", err)
+		case event := <-monitorClient.EventChannel():
+			if event.EventType() == monitor.EventTypeInitialStateTransfer {
+				logrus.Info("Monitor started")
+			}
+
+			for _, entity := range event.Entities() {
+				conn, ok := entity.(*connection.Connection)
+				if !ok || conn.GetId() != currentConn.GetId() {
+					continue
+				}
+				if conn.Context.DnsConfig != nil {
+					updateCoreFile(corefile, conn.Context.DnsConfig)
+				}
+				switch event.EventType() {
+				case monitor.EventTypeInitialStateTransfer, monitor.EventTypeUpdate:
+					if !proto.Equal(conn, currentConn) {
+						logrus.Infof("Connection updated: %v", conn)
+						currentConn = conn
+					}
+				case monitor.EventTypeDelete:
+					logrus.Info("Connection closed")
+					return
+				}
+			}
+		}
+	}
+}
+
+func updateCoreFile(corefile Corefile, config *connectioncontext.DNSConfig) {
+	forwards := strings.Join(config.DnsServerIps, " ")
+	corefile.Remove(".:53")
```

Comment:
Naively reading this, it looks like when we update the CoreFile we remove all existing DNS config and replace it with the DNSConfig from the latest connection?  Is this so?  What about the original DNS config from K8s?  What about the DNSConfig from the previous NSM Connections?  How do we parallelize amongst them?

Reply:
> we remove all existing DNS config and replace it with the DNSConfig from the latest connection? 
Yes, you are right. I planning to add merging between configs in next commit.
> What about the original DNS config from K8s?
At this moment we not changing file `resolv.conf`. We just changing the corefile for coredns siedecar
So if our local coreDNS server is not know about the host, then default k8s-dns service will trying to resolve host(or some else from the `resolve.conf`). 
Is this behaviour wrong?
>What about the DNSConfig from the previous NSM Connections? 
We'll store all configs.
>How do we parallelize amongst them?
I think it's part of coredns. I'll investigate and provide more detail answer.